orderBook.js

```diff
@@ -1,5 +1,45 @@
-function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+/* eslint-disable indent */
+const reconcileOrder = (existingBook, incomingOrder) => {
+
+    if (existingBook.length === 0) {
+        existingBook.push(incomingOrder)
+
+    } else {
+
+        const matchingOrders = existingBook.find(order => order.type !== incomingOrder.type && order.price === incomingOrder.price)
```

Comment:
Since find only returns the first one that matches, `matchingOrder` is a better name. `matchingOrders` makes it sound like an array.